pkg/operator/targetconfigcontroller/targetconfigcontroller.go

```diff
@@ -412,6 +445,113 @@ func ensureLocalhostRecoverySAToken(ctx context.Context, client coreclientv1.Cor
 	return err
 }
 
+func ensureMachineKubeconfigs(ctx context.Context, client coreclientv1.CoreV1Interface, infrastructureLister configv1.InfrastructureLister, recorder events.Recorder) error {
```

Comment:
this doesn't rely on any state in the observed config. Please move to a different control loop.

Reply:
ok